src/components/Signup/ConfirmForm.tsx

```diff
@@ -0,0 +1,97 @@
+import { yupResolver } from "@hookform/resolvers/yup";
+import { AuthError, confirmSignUp, resendSignUpCode } from "aws-amplify/auth";
+import { Field, Form } from "components/form";
+import { GENERIC_ERROR_MESSAGE } from "constants/common";
+import { useErrorContext } from "contexts/ErrorContext";
+import { useState } from "react";
+import { UseFormReturn, useForm } from "react-hook-form";
+import { requiredString } from "schemas/string";
+import { object } from "yup";
+import { CodeRecipientEmail, StateSetter } from "./types";
+
+type Props = {
+  codeRecipientEmail: CodeRecipientEmail;
+  setSignupState: StateSetter;
+  classes?: string;
+};
+export default function ConfirmForm(props: Props) {
+  const { handleError } = useErrorContext();
+  const [isRequestingNewCode, setIsRequestingNewCode] = useState(false);
+  const methods = useForm({
+    resolver: yupResolver(object({ code: requiredString })),
+  });
+  const {
+    handleSubmit,
+    formState: { isSubmitting },
+  } = methods;
+
+  type FV = typeof methods extends UseFormReturn<infer U> ? U : never;
+
+  return (
+    <Form
+      className={`${props.classes ?? ""} grid`}
+      disabled={isSubmitting || isRequestingNewCode}
+      methods={methods}
+      onSubmit={handleSubmit(async (fv) => {
+        try {
+          const result = await confirmSignUp({
+            username: props.codeRecipientEmail.raw,
+            confirmationCode: fv.code,
+          });
+
+          if (
+            result.nextStep.signUpStep !== "DONE" ||
+            !result.isSignUpComplete
+          ) {
+            return handleError("Code confirmation failed");
+          }
+
+          props.setSignupState("success");
+        } catch (err) {
+          const message =
+            err instanceof AuthError ? err.message : GENERIC_ERROR_MESSAGE;
+          handleError(message);
```

Comment:
```suggestion
          handleError(err, message);
```
We should still log the original error.

Reply:
I think only unexpected errors should be captured by sentry. but looking at current `handleError` seems like all sorts of error is captured even though some errors are made and can be corrected by user e.g. incorrect confirmation code, signing-up with taken email. 

Perhaps `handleError` would need to distinguish between unexpected (app failed) and expected (user failed)
e.g.
```
catch (err) {
if (err instanceof AuthError){
   return showModal(Error, err.message)
}
handleUnexpectedError(error)
}
```